tests/e2e/network/ensureloadbalancer.go

```diff
@@ -740,6 +744,315 @@ var _ = Describe("Ensure LoadBalancer", FlakeAttempts(3), Label(utils.TestSuiteL
 	})
 })
 
+var _ = Describe("EnsureLoadBalancer should not update any resources when service config is not changed", FlakeAttempts(3), func() {
+	basename := testBaseName
+
+	var cs clientset.Interface
+	var ns *v1.Namespace
+	var tc *utils.AzureTestClient
+	var deployment *appsv1.Deployment
+
+	labels := map[string]string{
+		"app": testServiceName,
+	}
+	ports := []v1.ServicePort{{
+		Port:       serverPort,
+		TargetPort: intstr.FromInt(serverPort),
+	}}
+
+	BeforeEach(func() {
+		var err error
+		cs, err = utils.CreateKubeClientSet()
+		Expect(err).NotTo(HaveOccurred())
+
+		ns, err = utils.CreateTestingNamespace(basename, cs)
+		Expect(err).NotTo(HaveOccurred())
+
+		tc, err = utils.CreateAzureTestClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		utils.Logf("Creating deployment %s", testDeploymentName)
+		deployment = createServerDeploymentManifest(testDeploymentName, labels)
+		_, err = cs.AppsV1().Deployments(ns.Name).Create(context.TODO(), deployment, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		err := cs.AppsV1().Deployments(ns.Name).Delete(context.TODO(), testDeploymentName, metav1.DeleteOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		err = utils.DeleteNamespace(cs, ns.Name)
+		Expect(err).NotTo(HaveOccurred())
+
+		cs = nil
+		ns = nil
+		tc = nil
+	})
+
+	It("should respect service with various configurations", func() {
+		By("Creating a service and expose it")
+		serviceDomainNamePrefix := testServiceName + string(uuid.NewUUID())
+		annotation := map[string]string{
+			consts.ServiceAnnotationDNSLabelName:                       serviceDomainNamePrefix,
+			consts.ServiceAnnotationAzurePIPTags:                       "Tag_1=t1, Tag_2=t2",
+			consts.ServiceAnnotationLoadBalancerIdleTimeout:            "20",
+			consts.ServiceAnnotationLoadBalancerHealthProbeProtocol:    "HTTP",
+			consts.ServiceAnnotationLoadBalancerHealthProbeRequestPath: "/healthtz",
+			consts.ServiceAnnotationLoadBalancerHealthProbeInterval:    "10",
+			consts.ServiceAnnotationLoadBalancerHealthProbeNumOfProbe:  "8",
+		}
+
+		if strings.EqualFold(os.Getenv(utils.LoadBalancerSkuEnv), string(network.PublicIPAddressSkuNameStandard)) {
+			// Routing preference is only supported in standard public IPs
+			annotation[consts.ServiceAnnotationIPTagsForPublicIP] = "RoutingPreference=Internet"
+		}
+
+		ip := createAndExposeDefaultServiceWithAnnotation(cs, testServiceName, ns.Name, labels, annotation, ports)
+		service, err := cs.CoreV1().Services(ns.Name).Get(context.TODO(), testServiceName, metav1.GetOptions{})
+		defer func() {
+			By("Cleaning up")
+			err := utils.DeleteServiceIfExists(cs, ns.Name, testServiceName)
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		By("Retrieving etags from resources")
```

Comment:
There tests create differente services and then: retrieve etags, add a dummy annotation, update services, check if etags are not changed. Is it possible to reuse code for the steps except service creation?

Reply:
sure, thx